src/Uno.Core/Collections/PairExtensions.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Collections;
+using Uno.Extensions;
+using Uno.Threading;
+
+namespace Uno.Core.Collections
+{
+	public static class EnumerableExtensions2
+	{
+		public static IEnumerable<Pair<object>> Pair(this IEnumerable xItems, IEnumerable yItems)
+		{
+			return Pair(xItems.Cast<object>(), yItems.Cast<object>());
+		}
+
+		public static IEnumerable<Pair<T>> Pair<T>(this IEnumerable<T> xItems, IEnumerable<T> yItems)
+		{
+			var xEnumerator = xItems.GetEnumerator();
+			var yEnumerator = yItems.GetEnumerator();
+
+			var xSuccess = xEnumerator.MoveNext();
+			var ySuccess = yEnumerator.MoveNext();
+
+			while (xSuccess && ySuccess)
+			{
+				yield return new Pair<T>(xEnumerator.Current, yEnumerator.Current);
+
+				xSuccess = xEnumerator.MoveNext();
+				ySuccess = yEnumerator.MoveNext();
+			}
+
+			if (xSuccess || ySuccess)
+			{
+				throw new ArgumentException("xItems && yItems not same length");
+			}
+		}
+
+		public static IList<T> ToLazyList<T>(this IEnumerable<T> items)
+		{
+			return new LazyList<T>(items.ToList);
+		}
+	}
+
+	public static class ListExtensions
+	{
+
+		/// <summary>
+		/// Adapts a list of type T into a list of type U
+		/// </summary>
+		/// <typeparam name="T">The type to adapt.</typeparam>
+		/// <typeparam name="U">The target type.</typeparam>
+		/// <param name="items">The list to adapt</param>
+		/// <param name="from">The function used to adapt a U into a T.</param>
+		/// <param name="to">The function used to adapt a T into a U.</param>
+		/// <returns>A adapted list of the target type.</returns>
+		public static IList<U> Adapt<T, U>(this IList<T> items, Func<U, T> from, Func<T, U> to)
+		{
+			return new ListAdapter<T, U>(items, from, to);
+		}
+
+		public static IList<U> Adapt<T, U>(this IList<T> items)
+		{
+			return Adapt(items, Funcs<U, T>.Convert, Funcs<T, U>.Convert);
+		}
+	}
+
+	public static class CollectionsExtensions
+	{
+		public static ICollection<U> Adapt<T, U>(this ICollection<T> collection)
+		{
+			return Adapt(collection, Funcs<U, T>.Convert, Funcs<T, U>.Convert);
+		}
+
+		/// <summary>
+		/// Adapts a collection of type T into a collection of type U
+		/// </summary>
+		/// <typeparam name="TSource">The type to adapt.</typeparam>
+		/// <typeparam name="TTarget">The target type.</typeparam>
+		/// <param name="collection">The collection to adapt</param>
+		/// <param name="from">The function used to adapt a U into a T.</param>
+		/// <param name="to">The function used to adapt a T into a U.</param>
+		/// <returns>A adapted collection of the target type.</returns>
+		public static ICollection<TTarget> Adapt<TSource, TTarget>(this ICollection<TSource> collection, Func<TTarget, TSource> from, Func<TSource, TTarget> to)
```

Comment:
```suggestion
		/// <summary>
		/// Adapts a list of type <typeparamref name="TSource"/> into a list of type <typeparamref name="TTarget"/>.
		/// </summary>
		/// <typeparam name="TSource">The type to adapt.</typeparam>
		/// <typeparam name="TTarget">The target type.</typeparam>
		/// <param name="items">The list to adapt</param>
		/// <param name="from">The function used to adapt a TTarget into a TSource.</param>
		/// <param name="to">The function used to adapt a TSource into a TTarget.</param>
		/// <returns>A adapted list of the target type.</returns>
		public static IList<TTarget> Adapt<TSource, TTarget>(this IList<TSource> items, Func<TTarget, TSource> from, Func<TSource, TTarget> to)
		{
			return new ListAdapter<TSource, TTarget>(items, from, to);
		}

		public static IList<U> Adapt<T, U>(this IList<T> items)
		{
			return Adapt(items, Funcs<U, T>.Convert, Funcs<T, U>.Convert);
		}
	}

	public static class CollectionsExtensions
	{
		public static ICollection<TTarget> Adapt<TSource, TTarget>(this ICollection<TSource> collection)
		{
			return Adapt(collection, Funcs<TTarget, TSource>.Convert, Funcs<TSource, TTarget>.Convert);
		}

		/// <summary>
		/// Adapts a collection of type <typeparamref name="TSource"/> into a collection of type <typeparamref name="TTarget"/>.
		/// </summary>
		/// <typeparam name="TSource">The type to adapt.</typeparam>
		/// <typeparam name="TTarget">The target type.</typeparam>
		/// <param name="collection">The collection to adapt</param>
		/// <param name="from">The function used to adapt a TTarget into a TSource.</param>
		/// <param name="to">The function used to adapt a TSource into a TTarget.</param>
		/// <returns>A adapted collection of the target type.</returns>
		public static ICollection<TTarget> Adapt<TSource, TTarget>(this ICollection<TSource> collection, Func<TTarget, TSource> from, Func<TSource, TTarget> to)
```

Reply:
This is legacy code, no particular need to touch this at this point :)